src/main/java/reactor/netty/http/MicrometerHttpMetricsRecorder.java

```diff
@@ -19,28 +19,23 @@
 import io.micrometer.core.instrument.DistributionSummary;
 import io.micrometer.core.instrument.Timer;
 import io.netty.util.internal.PlatformDependent;
-import reactor.netty.Metrics;
 import reactor.netty.channel.MeterKey;
 import reactor.netty.channel.MicrometerChannelMetricsRecorder;
 
-import java.net.SocketAddress;
 import java.util.concurrent.ConcurrentMap;
 
 import static reactor.netty.Metrics.DATA_RECEIVED;
 import static reactor.netty.Metrics.DATA_RECEIVED_TIME;
 import static reactor.netty.Metrics.DATA_SENT;
 import static reactor.netty.Metrics.DATA_SENT_TIME;
 import static reactor.netty.Metrics.ERRORS;
-import static reactor.netty.Metrics.REGISTRY;
-import static reactor.netty.Metrics.REMOTE_ADDRESS;
 import static reactor.netty.Metrics.RESPONSE_TIME;
-import static reactor.netty.Metrics.URI;
 
 /**
  * @author Violeta Georgieva
  * @since 0.9
  */
-public class MicrometerHttpMetricsRecorder extends MicrometerChannelMetricsRecorder implements HttpMetricsRecorder {
+public abstract class MicrometerHttpMetricsRecorder extends MicrometerChannelMetricsRecorder implements HttpMetricsRecorder {
```

Comment:
I _think_ that is binary compatible, the `abstract` + removal of overrides in this base class, but can you confirm JAPICMP doesn't find any issue here?

Reply:
There is an issue if this class is inherited.
Fixed